### PR TITLE
(feat) Add samples to metadata

### PR DESCRIPTION
**Pull request by mttrbrts**
Adds feature to clause-core metadata and template classes to support sample text. 

- Samples can be provided in an explicit locale. Locale tags conform to the [IETF Language Tags format (BCP 47)](https://tools.ietf.org/html/bcp47)
- Sample files should appear in the root folder of the template package and follow the naming convention 
  - `sample.txt`
  - `sample_TAG.txt`, where `TAG` is an IETF language tag, e.g. `sample_fr.txt`, `sample_zh-cmn.txt`.
- Where a sample locale is not provided it assumes a `default` locale. Default samples should be placed in `sample.txt` 

Note that samples provided in locales that do not match a given template are not yet supported.

### CODE REVIEW

File: packages/cicero-core/lib/template.js [hunk 1]
```diff
@@ -55,15 +61,16 @@ class Template {
      * retrieve instances from {@link Template.fromArchive}.
      * @param {object} packageJson  - the JS object for package.json
      * @param {String} readme  - the readme in markdown for the clause (optional)
+     * @param {object} samples - the sample text for the template in different locales (optional)
```
Comment (dselman) [inline packages/cicero-core/lib/template.js]: Not optional.

File: packages/cicero-core/lib/metadata.js [hunk 2]
```diff
@@ -79,6 +84,31 @@ class Metadata {
...
+
+    /**
+     * Returns the sample for this template in the given locale. This may be null.
+     * If no locale is specified returns the default sample if it has been specified.
+     *
+     * @param {string} locale the IETF language code for the language
+     * @return {string} the sample file for the template in the given locale or null
+     */
+    getSample(locale) {
+        if(!locale && 'default' in this.samples){
```
Comment (dselman) [inline packages/cicero-core/lib/metadata.js]: Document magic key and format of keys.

File: packages/cicero-core/lib/template.js [hunk 3]
```diff
@@ -917,13 +984,34 @@ class Template {
...
+     * @private
+     */
+    setSamples(samples) {
+        this.metadata = new Metadata(this.metadata.getPackageJson(), this.metadata.getREADME(), samples);
+    }
+
+    /**
+     * Set a locale-specified sample within the Metadata
+     * @param {object} sample the samples for the template
+     * @param {string} locale the two letter ISO 639-1 code for the language
```
Comment (dselman) [inline packages/cicero-core/lib/template.js]: Update spec.

File: packages/cicero-core/lib/template.js (hunk 1 above)
Comment (mttrbrts) [reply packages/cicero-core/lib/template.js]: 👍

**Pull request merged**